src/OrchardCore/OrchardCore.Lucene.Core/QueryProviders/SimpleQueryStringQueryProvider.cs

```diff
@@ -17,7 +19,18 @@ public Query CreateQuery(ILuceneQueryService builder, LuceneQueryContext context
             var queryString = query["query"]?.Value<string>();
             var fields = query["fields"]?.Values<string>() ?? new string[0];
             var defaultOperator = query["default_operator"]?.Value<string>().ToLowerInvariant() ?? "or";
-            var weights = fields.ToDictionary(field => field, field => 1.0f);
+            var weight = 1.0f;
+            var weights = fields.ToDictionary(field => field, field => weight);
+
+            foreach(var field in fields)
+            {
+                if(field.Contains('^', StringComparison.Ordinal) && Single.TryParse(field.Split("^").Last(), out weight))
```

Comment:
@Skrypt 

Yes, for readability maybe better to specify `StringComparison.Ordinal` as this is what we want, even here it executes a few more lines of code (but no significant impact)

So just to know, did you re-add it because you saw some perf impact? Normally `Contains()` doesn't use any comparison type by default.

Note: Didn't check the version of `Split('^')` accepting a char ;)